lib/WasmReader/WasmBinaryReader.cpp

```diff
@@ -167,6 +183,35 @@ WasmBinaryReader::ReadFromModule()
      return wnLIMIT;
 }
 
+SectionCode
+WasmBinaryReader::SectionHeader()
+{
+    UINT len;
+    UINT sectionSize;
+    UINT idSize;
+
+    sectionSize = LEB128(len);
+    m_moduleState.count += len;
+    idSize = LEB128(len);
+    m_moduleState.count += len;
+    const char *sectionName = (char*)(m_pc);
+    m_pc += idSize;
+    m_moduleState.count += idSize;
+
```

Comment:
You are incrementing m_moduleState.count here, but it gets reset to 0 at the call-site of SectionHeader. I think you should reset the count here, and also set the size.